lib/stripe/webhook.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Stripe.Webhook do
+  @moduledoc """
+  Creates a Stripe Event from webhook's payload if signature is valid.
+
+  Use `construct_event/3` to verify the authenticity of a webhook request and
+  convert its payload into a `Stripe.Event` struct.
+
+    case Stripe.Webhook.construct_event(payload, signature, secret) do
+      {:ok, %Stripe.Event{} = event} ->
+        # Return 200 to Stripe and handle event
+      {:error, reason} ->
+        # Reject webhook by responding with non-2XX
+    end
+  """
+  @default_tolerance 300
+  @expected_scheme "v1"
+
+  def construct_event(payload, signature_header, secret, tolerance \\ @default_tolerance) do
+    case verify_header(payload, signature_header, secret, tolerance) do
+      :ok ->
+        {:ok, Poison.decode!(payload)}
```

Comment:
Sorry about that, I fixed the return value (yes, it must return a `Stripe.Event`) but got sidetracked at work. Changes pushed now.

I only went with `Poison.decode!/1` because other parts of this library do the same. I think that’s a reasonable assumption because invalid JSON from Stripe would be exceptional.